GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -25,44 +25,26 @@ public class GitHubUpgrader : IProductUpgrader
         private const string GVFSAssetId = "GVFS";
         private const string GitInstallerFileNamePrefix = "Git-";
         private const int RepoMountFailureExitCode = 17;
-        private const string ToolsDirectory = "Tools";
-        private static readonly string UpgraderToolName = GVFSPlatform.Instance.Constants.GVFSUpgraderExecutableName;
-        private static readonly string UpgraderToolConfigFile = UpgraderToolName + ".config";
-        private static readonly string[] UpgraderToolAndLibs =
-            {
-                UpgraderToolName,
-                UpgraderToolConfigFile,
-                "GVFS.Common.dll",
-                "GVFS.Platform.Windows.dll",
-                "Microsoft.Diagnostics.Tracing.EventSource.dll",
-                "netstandard.dll",
-                "System.Net.Http.dll",
-                "Newtonsoft.Json.dll"
-            };
 
-        private Version installedVersion;
         private Version newestVersion;
         private Release newestRelease;
         private PhysicalFileSystem fileSystem;
         private ITracer tracer;
+        private LocalUpgraderServices localUpgradeServices;
+        private Version installedVersion;
 
         public GitHubUpgrader(
             string currentVersion,
             ITracer tracer,
             GitHubUpgraderConfig upgraderConfig)
-            : this(currentVersion, tracer)
         {
             this.Config = upgraderConfig;
-        }
 
-        public GitHubUpgrader(string currentVersion, ITracer tracer)
-        {
             this.installedVersion = new Version(currentVersion);
             this.fileSystem = new PhysicalFileSystem();
             this.tracer = tracer;
 
-            string upgradesDirectoryPath = ProductUpgraderInfo.GetUpgradesDirectoryPath();
-            this.fileSystem.CreateDirectory(upgradesDirectoryPath);
+            this.localUpgradeServices = new LocalUpgraderServices(tracer);

```

Comment:
You create a new `PhysicalFileSystem` above, but then also create a new one inside `LocalUpgraderServices()`. Perhaps supply one in the constructor?